modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -579,6 +588,251 @@ public void preprocessOpenAPI(OpenAPI openAPI) {
                 }
             }
         }
+
+        if (useOneOfInterfaces && openAPI.getComponents() != null){
```

Comment:
Since this method calls `super.preprocessOpenAPI`, I don't think you should need to include this block at all. Is there a specific reason why you added it here?

Reply:
I added it because SpringCodegenTest.testAdditionalPropertiesPutForConfigValues and 
SpringCodegenTest.testInitialConfigValues fails with NPE  otherway.